src/i_oalsound.c

```diff
@@ -0,0 +1,662 @@
+    }
+
+    oal->SOFT_source_spatialize = (alIsExtensionPresent("AL_SOFT_source_spatialize") == AL_TRUE);
+    oal->EXT_EFX = (alcIsExtensionPresent(oal->device, "ALC_EXT_EFX") == AL_TRUE);
+    oal->EXT_SOURCE_RADIUS = (alIsExtensionPresent("AL_EXT_SOURCE_RADIUS") == AL_TRUE);
+    oal->active = malloc(sizeof(*oal->active) * MAX_CHANNELS);
+    InitDeferred();
+    ResetParams();
+
+    return true;
+}
+
+boolean I_OAL_ReinitSound(boolean use_3d)
+{
+    LPALCRESETDEVICESOFT alcResetDeviceSOFT = NULL;
+    ALCint attribs[OAL_NUM_ATTRIBS];
+
+    if (!oal)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: OpenAL not initialized.\n");
+        return false;
+    }
+
+    if (alcIsExtensionPresent(oal->device, "ALC_SOFT_HRTF") != AL_TRUE)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: Extension not present.\n");
+        return false;
+    }
+
+    alcResetDeviceSOFT = FUNCTION_CAST(LPALCRESETDEVICESOFT,
+                                       alGetProcAddress("alcResetDeviceSOFT"));
+
+    if (!alcResetDeviceSOFT)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: Function address not found.\n");
+        return false;
+    }
+
+    GetAttribs(use_3d, attribs);
+
+    if (alcResetDeviceSOFT(oal->device, attribs) != AL_TRUE)
+    {
+        fprintf(stderr, "I_OAL_ReinitSound: Error resetting device.\n");
+        I_OAL_ShutdownSound();
+        return false;
+    }
+
+    ResetParams();
+
+    return true;
+}
+
+boolean I_OAL_AllowReinitSound(void)
+{
+    if (!oal)
+    {
+        return false;
+    }
+
+    // alcResetDeviceSOFT() is part of the ALC_SOFT_HRTF extension.
+    return (alcIsExtensionPresent(oal->device, "ALC_SOFT_HRTF") == AL_TRUE);
+}
+
+boolean I_OAL_CacheSound(ALuint *buffer, ALenum format, const byte *data,
+                         ALsizei size, ALsizei freq)
+{
+    if (!oal)
+    {
+        return false;
+    }
+
+    alGetError();
+    alGenBuffers(1, buffer);
+    if (alGetError() != AL_NO_ERROR)
+    {
+        fprintf(stderr, "I_OAL_CacheSound: Error creating buffers.\n");
+        return false;
+    }
+
+    alGetError();
+    alBufferData(*buffer, format, data, size, freq);
+    if (alGetError() != AL_NO_ERROR)
+    {
+        fprintf(stderr, "I_OAL_CacheSound: Error buffering data.\n");
+        return false;
+    }
+
+    if (oal->num_buffers == oal->num_buffers_mem)
+    {
+        ALuint *new_buffers;
+        oal->num_buffers_mem += NUMSFX;
+        new_buffers = realloc(oal->buffers, sizeof(ALuint) * (oal->num_buffers_mem));
+        if (new_buffers == NULL)
+        {
+            fprintf(stderr, "I_OAL_CacheSound: Error allocating buffers.\n");
+            return false;
+        }
+        oal->buffers = new_buffers;
+    }
+
+    oal->buffers[oal->num_buffers] = *buffer;
+    oal->num_buffers++;
+
+    return true;
+}
+
+boolean I_OAL_StartSound(int channel, ALuint buffer, int pitch)
+{
+    if (!oal)
+    {
+        return false;
+    }
+
+    alDeferUpdatesSOFT();
+
+    alGetError();
```

Comment:
The numer of calls to `alGetError()` is quite excessive in this part of the code. Any performance concerns?

Reply:
This is left over from debugging. Fixed by https://github.com/fabiangreffrath/woof/pull/1118/commits/3ebca0d8b40297467138063f497196d54825371b